packages/aragon-wrapper/src/cache/index.js

```diff
@@ -9,34 +9,19 @@ import { concat } from 'rxjs/observable/concat'
 export default class Cache {
   constructor (prefix) {
     this.prefix = prefix
-
-    // Set up cache DB
-    this.db = localforage.createInstance({
-      driver: [localforage.INDEXEDDB, localforage.LOCALSTORAGE],
-      name: `localforage/${this.prefix}`
-    })
   }
 
   async init () {
     // Set up the changes observable
     this.changes = new Subject()
 
-    try {
-      // Make sure localforage has settled down and is not waiting for anything else
-      // before possibly setting new drivers
-      await this.db.ready()
-    } catch (err) {
-      // If localforage isn't able to automatically connect to a driver
-      // due to lack of support in the environment (e.g. node),
-      // use an in-memory driver instead
-      // TODO: this doesn't provide an persistent cache for node
-      if (this.db.driver() === null) {
-        await this.db.defineDriver(memoryStorageDriver)
-        await this.db.setDriver(memoryStorageDriver._driver)
-      }
-
-      await this.db.ready()
-    }
+    await localforage.defineDriver(memoryStorageDriver)
```

Comment:
@0x6431346e Oops, didn't see this yesterday!

What do you think about 8263657?

If we create the localforage instance first, we don't have to redefine the memory storage driver on subsequent instantiations (since the driver definitions are global to all localforage instances). I think it's also better to avoid defining the in-memory driver unless we absolutely have to.

Reply:
Oh, I see. Not as pretty, but makes sense if we optimize for performance :smile: